enigma/src/main/java/cuchaz/enigma/translation/mapping/serde/RawEntryMapping.java

```diff
@@ -1,23 +1,23 @@
 package cuchaz.enigma.translation.mapping.serde;
 
-import cuchaz.enigma.translation.mapping.AccessModifier;
-import cuchaz.enigma.translation.mapping.EntryMapping;
-
 import java.util.ArrayList;
 import java.util.List;
 
+import cuchaz.enigma.translation.mapping.AccessModifier;
+import cuchaz.enigma.translation.mapping.EntryMapping;
+
 public final class RawEntryMapping {
 	private final String targetName;
 	private final AccessModifier access;
-	private List<String> javadocs = new ArrayList<>();
+	private final List<String> javadocs = new ArrayList<>();
 
 	public RawEntryMapping(String targetName) {
 		this(targetName, null);
 	}
 
 	public RawEntryMapping(String targetName, AccessModifier access) {
 		this.access = access;
-		this.targetName = targetName;
+		this.targetName = targetName != null && !targetName.equals("-") ? targetName : null;
```

Comment:
Hmm, what is the `-` check for?

Reply:
According to the PR description, it's used as a stand-in for a deobfuscated name in the Enigma format, to allow writing the access modifier component without a deobfuscated name.

Reply:
https://docs.oracle.com/javase/specs/jvms/se7/html/jvms-4.html#jvms-4.2.2
I suggest using one of `. ; [ /` to indicate an unchanged target name to avoid potential conflicts, preferably just `.`

Reply:
Is that necessary? These are deobf names, "-" is not a valid Java identifier character so you can't set it from the enigma UI and if you were to edit the file manually and have it remap correctly, you can't use methods with "-" in them from Java (nor probably any other JVM language). For all intents and purposes I think this is not really an issue (and I'd say "-" is more fitting as a placeholder to denote "no value" than any of these). I could make it "[none]" or something like that but as I said I don't think it'd be a problem.

Reply:
Nitpick: dashes are usable in Kotlin if you escape the name with backticks (though that's *extremely* cursed)